src/tests/intg/test_kcm.py

```diff
@@ -585,3 +627,40 @@ def test_kcm_secrets_quota(setup_for_kcm_sec,
     princ = "%s%d" % ("kcmtest", MAX_SECRETS)
     out, _, _ = testenv.k5util.kinit(princ, princ)
     assert out != 0
+
+
+def test_kcm_renewals(setup_for_kcm_renewals_secdb):
+    """
+    Test that basic KCM renewal works
+    """
+    testenv = setup_for_kcm_renewals_secdb
+    testenv.k5kdc.add_principal("user1", "Secret123")
+
+    ok = testenv.k5util.has_principal("user1@KCMTEST")
+    assert ok is False
+    nprincs = testenv.k5util.num_princs()
+    assert nprincs == 0
+
+    # Renewal is only performed after half of lifetime exceeded,
+    # see kcm_renew_all_tgts()
+    options = ["-r", "15s", "-l", "15s"]
+    out, _, _ = testenv.k5util.kinit_with_opts("user1",
+                                               "Secret123",
+                                               options)
+    assert out == 0
+    nprincs = testenv.k5util.num_princs()
+    assert nprincs == 1
+
+    timestr_fmt = "%m/%d/%y %H:%M:%S"
+    initial_times = testenv.k5util.list_times()
+
+    # Wait for renewal to trigger once, after renew interval
+    time.sleep(15)
```

Comment:
It's not ideal, but the same approach is used for other integration tests.

Reply:
`libfaketime` looks interesting. It would be good if it helps to reduce the test run time. But it is probably out of scope of this work, unless you'd like to look at it.